File: algorithm/problem/timeConversion/answer.py
def timeConversion(s):
  hh, mm, ss = map(int, s[:-2].split(':'))
  aa = s[-2:]

  hh = hh % 12 + (aa.upper() == 'PM') * 12
  print(('%02d:%02d:%02d') % (hh, mm, ss))

  mTime = "{0:02d}:{1:02d}:{2:02d}".format(hh, mm, ss)
  return mTime

File: algorithm/problem/timeConversion/answer.java

public class Solution {

    static String timeConversion(String s) {
      String[] time = s.split(":");
      String hour = time[0];
      String minit = time[1];
      String second = time[2];
      
      // parse strig to int
      int hourInt = Integer.parseInt(hour);
      
      // 12:00:00PM on a 12-hour clock is 12:00:00 on a 24-hour clock.
      if (second.contains("PM") && hourInt != 12) {
        hourInt += 12;
        hour = String.valueOf(hourInt);
      }

      // 12:00:00AM on a 12-hour clock is 00:00:00 on a 24-hour clock.
      if (second.contains("AM") && hourInt == 12) {
        hour = "00";
      }
      
      // replace PM/AM
      second  = second.replaceAll("[^\\d.]", "");
      String convertedTime = hour + ":" + minit + ":" + second;
      return convertedTime;
    }
}
